src/DotNetWorker/GrpcWorker.cs

```diff
@@ -2,41 +2,99 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Threading;
 using System.Threading.Channels;
 using System.Threading.Tasks;
+using Grpc.Core;
 using Microsoft.Azure.Functions.Worker.Context;
 using Microsoft.Azure.Functions.Worker.Context.Features;
 using Microsoft.Azure.Functions.Worker.Grpc.Messages;
 using Microsoft.Azure.Functions.Worker.Invocation;
 using Microsoft.Azure.Functions.Worker.OutputBindings;
-using static Microsoft.Azure.Functions.Worker.Grpc.Messages.StatusResult.Types;
+using Microsoft.Extensions.Options;
+using static Microsoft.Azure.Functions.Worker.Grpc.Messages.FunctionRpc;
 using MsgType = Microsoft.Azure.Functions.Worker.Grpc.Messages.StreamingMessage.ContentOneofCase;
 
 
 namespace Microsoft.Azure.Functions.Worker
 {
     internal class GrpcWorker : IWorker
     {
+        private readonly ChannelReader<StreamingMessage> _outputReader;
         private readonly ChannelWriter<StreamingMessage> _outputWriter;
+
         private readonly IFunctionsApplication _application;
+        private readonly FunctionRpcClient _rpcClient;
         private readonly IInvocationFeaturesFactory _invocationFeaturesFactory;
         private readonly IOutputBindingsInfoProvider _outputBindingsInfoProvider;
         private readonly IMethodInfoLocator _methodInfoLocator;
+        private readonly IOptions<GrpcWorkerStartupOptions> _startupOptions;
+        private Task? _writerTask;
+        private Task? _readerTask;
 
-        public GrpcWorker(IFunctionsApplication application, FunctionsHostOutputChannel outputChannel, IInvocationFeaturesFactory invocationFeaturesFactory,
-            IOutputBindingsInfoProvider outputBindingsInfoProvider, IMethodInfoLocator methodInfoLocator)
+        public GrpcWorker(IFunctionsApplication application, FunctionRpcClient rpcClient, FunctionsHostOutputChannel outputChannel, IInvocationFeaturesFactory invocationFeaturesFactory,
```

Comment:
I don't like how big this is getting; will follow-up later with some refactoring.